gnovm/pkg/doc/json_doc.go

```diff
@@ -113,32 +143,126 @@ func (d *Documentable) WriteJSONDocumentation() (*JSONDocumentation, error) {
 	for _, fun := range pkg.Funcs {
 		jsonDoc.Funcs = append(jsonDoc.Funcs, &JSONFunc{
 			Name:      fun.Name,
-			Signature: mustFormatNode(d.pkgData.fset, fun.Decl),
+			Crossing:  isCrossing(fun),
+			Signature: mustFormatNode(d.pkgData.fset, fun.Decl, opt.Source, file),
 			Doc:       string(pkg.Markdown(fun.Doc)),
 			Params:    d.extractJSONFields(fun.Decl.Type.Params),
 			Results:   d.extractJSONFields(fun.Decl.Type.Results),
 		})
 	}
 
 	for _, typ := range pkg.Types {
+		// Set typeSpec to the ast.TypeSpec within the declaration that defines the symbol.
+		var typeSpec *ast.TypeSpec
+		for _, spec := range typ.Decl.Specs {
+			tSpec := spec.(*ast.TypeSpec) // Must succeed
+			if typ.Name == tSpec.Name.Name {
+				typeSpec = tSpec
+				break
+			}
+		}
+		if typeSpec == nil || typeSpec.Type == nil {
+			// We don't expect this
+			continue
+		}
+		typeExpr := typeSpec.Type
+
+		// De-parenthesize
+		for {
+			if t, ok := typeExpr.(*ast.ParenExpr); ok {
+				typeExpr = t.X
+			} else {
+				break
+			}
+		}
+
+		kind := ""
+		var methods []*JSONFunc
+		var fields []*JSONField
+
+		switch t := typeExpr.(type) {
+		case *ast.StructType:
+			kind = structKind
+			// TODO: Anonymous fields.
+			fields = d.extractJSONFields(t.Fields)
+		case *ast.InterfaceType:
+			kind = interfaceKind
+			for _, iMethod := range t.Methods.List {
+				fun, ok := iMethod.Type.(*ast.FuncType)
+				if !ok {
+					// We don't expect this
+					continue
```

Comment:
We should handle embedded interfaces, like

```
type WriteCloser interface {
	io.Writer
	Close() error
}
```

Do you think we can have it directly as JSONFunc?

Maybe it is better to have its own type, like `JSONInterfaceMethod`.

Reply:
I can make its own type like the following. What should the entry for the embedded type look like? 
```
type JSONInterfaceMethod struct {
	Name      string       `json:"name"`
	Signature string       `json:"signature"`
	Doc       string       `json:"doc"` // markdown
	Params    []*JSONField `json:"params"`
	Results   []*JSONField `json:"results"`
}
```